--- a/drivers/mfd/Kconfig
+++ b/drivers/mfd/Kconfig
@@ -2228,5 +2228,14 @@ config MFD_RSMU_SPI
 	  Additional drivers must be enabled in order to use the functionality
 	  of the device.
 
+config MFD_LANDISK_R8C
+	tristate "Renesas R8C MCU on I-O DATA LAN DISK series"
+	depends on SERIAL_DEV_BUS
+	select MFD_CORE
+	help
+	  This option enables common support for the Renesas R8C MCU on
+	  I-O DATA LAN DISK series NAS. It has several functions, such as
+	  LEDs, keys, temperature sensor, fan controller, and others.
+
 endmenu
 endif
--- a/drivers/mfd/Makefile
+++ b/drivers/mfd/Makefile
@@ -278,3 +278,4 @@ rsmu-i2c-objs			:= rsmu_core.o rsmu_i2c.
 rsmu-spi-objs			:= rsmu_core.o rsmu_spi.o
 obj-$(CONFIG_MFD_RSMU_I2C)	+= rsmu-i2c.o
 obj-$(CONFIG_MFD_RSMU_SPI)	+= rsmu-spi.o
+obj-$(CONFIG_MFD_LANDISK_R8C)	+= landisk-r8c.o
